src/order_cmd.cpp

```diff
@@ -1845,13 +1845,23 @@ void DeleteVehicleOrders(Vehicle *v, bool keep_orderlist, bool reset_order_indic
 
 /**
  * Clamp the service interval to the correct min/max. The actual min/max values
- * depend on whether it's in percent or days.
- * @param interval proposed service interval
- * @return Clamped service interval
+ * depend on whether it's in days, minutes, or percent.
+ * @param interval The proposed service interval.
+ * @param ispercent Whether the interval is a percent.
+ * @return The service interval clamped to use the chosen units.
  */
 uint16_t GetServiceIntervalClamped(uint interval, bool ispercent)
 {
-	return ispercent ? Clamp(interval, MIN_SERVINT_PERCENT, MAX_SERVINT_PERCENT) : Clamp(interval, MIN_SERVINT_DAYS, MAX_SERVINT_DAYS);
+	/* Service intervals are in percents. */
+	if (ispercent) return Clamp(interval, MIN_SERVINT_PERCENT, MAX_SERVINT_PERCENT);
+
+	bool realtime_units = (_game_mode == GM_MENU ? _settings_newgame.economy.use_realtime_units : _settings_game.economy.use_realtime_units);
```

Comment:
Why does this depend on `_game_mode`?

Reply:
This function is used in the Settings menu as a post_cb when changing default service intervals. When we change these in the main menu, we want to change the newgame settings, not those of the title game.